runtime_tests/build.sh

```diff
@@ -0,0 +1,88 @@
+#!/bin/bash
+# Copyright 2020 Google Inc. All Rights Reserved
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+if [ -z $1 ]; then
+  COMPILATION_LEVEL="SIMPLE"
+else
+  COMPILATION_LEVEL=$1
+fi
+
+# Get the location of the local compiler in this directory, if it exists.
+# If it doesn't, build it, then resume execution.
+LOCAL_COMPILER="$(dirname ..)/target/closure-compiler-1.0-SNAPSHOT.jar"
+if [ ! -f "$LOCAL_COMPILER" ]; then
+  echo -e "\nCompiler JAR not built. Building...\n" && yarn build:fast
+fi
+
+# Translate a relative filepath to an absolute one.
+abs_dirname() {
+  echo "$(cd "$(dirname "$1")" && pwd)/$(basename "$1")"
+}
+
+# Begin building runtime tests in the $TEST_DIR directory.
+echo -e "\nBuilding runtime tests..."
+TEST_DIR="test/com/google/javascript/jscomp/runtime_tests"
+
+# Get the absolute path of the test directory.
+ABS_PATH=$(abs_dirname "./$TEST_DIR")
+
+compileRuntimeTests(){
+  local -i i=0
+  for file in $@; do
+
+    # /path/to/file.ext -> /path/to/file
+    local file_base=$(echo $file | rev | cut -f 2- -d '.' | rev)
+    # /path/to/file -> file
+    local test_name=$(basename $file_base)
+    # /path/to/file.ext -> /path/to
+    local test_loc=$(dirname $file)
+
+    # Make sure the build directory exists.
+    mkdir -p $test_loc/build
+
+    # Echo a percentage progress indicator.
+    ((i += 1))
+    echo " $((100 * $i / $#))% | $test_name"
+
+    # Output the test file, which will be executed in JSDOM.
+    echo "
```

Comment:
nit: the preferred way to specify multi-line text in a bash script is to use a heredoc

https://en.wikipedia.org/wiki/Here_document#:~:text=In%20computing%2C%20a%20here%20document,it%20were%20a%20separate%20file.

```shell
cat > $test_loc/build/$test_name.html <<_eof_
HTML_TEXT_HERE
_eof_
```